tests/sentry/api/endpoints/test_project_app_store_connect_credentials.py

```diff
@@ -0,0 +1,100 @@
+from sentry.api.endpoints.project_app_store_connect_credentials import (
+    AppStoreUpdateCredentialsSerializer,
+)
+from sentry.utils import json
+
+
+class TestAppStoreUpdateCredentialsSerializer:
+    def test_validate_secrets_magic_object_true(self):
+        payload_json = """{
+            "appconnectPrivateKey": { "hidden-secret": true },
+            "itunesPassword":  { "hidden-secret": true }
+        }"""
+
+        payload = json.loads(payload_json)
+        serializer = AppStoreUpdateCredentialsSerializer(data=payload)
+        if not serializer.is_valid():
+            raise AssertionError(serializer.errors)
```

Comment:
```suggestion
            pytest.fail(serializer.errors)
```
according to the imaginary pytest rules, but consider this more nitpicking